lib/internal/modules/esm/fetch.js

```diff
@@ -0,0 +1,287 @@
+'use strict';
+const {
+  globalThis,
+  ArrayPrototypePush,
+  Promise,
+  PromisePrototypeThen,
+  PromiseResolve,
+  SafeMap,
+  StringPrototypeEndsWith,
+  StringPrototypeSlice,
+  StringPrototypeStartsWith,
+} = primordials;
+const {
+  Buffer: {
+    concat: BufferConcat
+  }
+} = globalThis;
+const {
+  ERR_NETWORK_IMPORT_DISALLOWED,
+  ERR_NETWORK_IMPORT_BAD_RESPONSE,
+} = require('internal/errors').codes;
+const { URL } = require('internal/url');
+const net = require('net');
+const path = require('path');
+
+/**
+ * @typedef CacheEntry
+ * @property {Promise<string> | string} resolvedHREF
+ * @property {Record<string, string>} headers
+ * @property {Promise<Buffer> | Buffer} body
+ */
+
+/**
+ * Only for GET requests, other requests would need new Map
+ * HTTP cache semantics keep diff caches
+ *
+ * Maps HREF to pending cache entry
+ * @type {Map<string, Promise<CacheEntry> | CacheEntry>}
+ */
+const cacheForGET = new SafeMap();
+
+
+/**
+ * Lazily load these to avoid snapshot issues
+ */
+let HTTPSAgent;
+function HTTPSGet(url, opts) {
+  const https = require('https');
+  HTTPSAgent ??= new https.Agent({
+    keepAlive: true
+  });
```

Comment:
Is there any security risk in doing so? Otherwise, that sounds like a good idea to me.

Reply:
I can't find Bradley's comment (I got an email notification for it), but it says the global agent should not be used because it could be tainted (which is what I was concerned about above).

Reply:
security risk is one thing, also this is a perf improvement generally, also mutating the agent isn't in the right timing since if your entrypoint is using an `import` it would run all the loading prior to mutating the global.